django/contrib/admin/utils.py

```diff
@@ -296,10 +296,14 @@ def lookup_field(name, obj, model_admin=None):
             attr = getattr(model_admin, name)
             value = attr(obj)
         else:
-            attr = getattr(obj, name)
+            attr = getattr(obj, name, None)
             if callable(attr):
                 value = attr()
             else:
+                if not attr:
```

Comment:
This guard would be True for any value of `attr` that evaluates to False (`''`, `0`, `[]`, etc), so I believe that we'd want instead to check for exactly the sentinel value we used in the `getattr` call.